Gordon360/Hubs/SignalRHub.cs

```diff
@@ -20,30 +20,80 @@ namespace Gordon360.Hubs
     [HubName("ChatHub")]
     public class ChatHub : Hub
     {
-        //public DirectMessageService _DirectMessageService = new DirectMessageService();
+        public async Task ManageGroupAsync(string connectionId, string groupName, string function)
+        {
+            if (connectionId == null)
+            {
+                throw new ArgumentNullException(nameof(connectionId));
+            }
+
+            if (groupName == null)
+            {
+                throw new ArgumentNullException(nameof(groupName));
+            }
+            
+
+            if (function == "add")
+            {
+                await Groups.Add(connectionId, groupName);
+
+            } else
+            {
+                Groups.Remove(connectionId, groupName);
```

Comment:
This was purposefully removed because according to the signalR documentation, it is better not to await the removal or deleting of things from groups.

Here's the reasoning: In general, you should not include await when calling the Groups.Remove method because the connection id that you are trying to remove might no longer be available. In that case, TaskCanceledException is thrown after the request times out. If your application must ensure that the user has been removed from the group before sending a message to the group, you can add await before Groups.Remove, and then catch the TaskCanceledException exception that might be thrown.

https://docs.microsoft.com/en-us/aspnet/signalr/overview/guide-to-the-api/working-with-groups

Reply:
Gotcha, that makes perfect sense. Thanks for the clarification!